rct229/rulesets/ashrae9012019/section11/section11rule6.py

```diff
@@ -0,0 +1,85 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.ruleset_model_factory import produce_ruleset_model_description
+from rct229.rulesets.ashrae9012019 import BASELINE_0
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_swh_uses_associated_with_each_building_segment import (
+    get_swh_uses_associated_with_each_building_segment,
+)
+from rct229.utils.jsonpath_utils import find_all, find_exactly_one_with_field_value
+
+
+class Section11Rule6(RuleDefinitionListIndexedBase):
+    """Rule 6 of ASHRAE 90.1-2019 Appendix G Section 11 (Service Water Heating)"""
+
+    def __init__(self):
+        super(Section11Rule6, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=False
+            ),
+            each_rule=Section11Rule6.RMDRule(),
+            index_rmd=BASELINE_0,
+            id="11-6",
+            description="Piping losses shall not be modeled.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, i",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule6.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False, BASELINE_0=True, PROPOSED=False
+                ),
+                each_rule=Section11Rule6.RMDRule.BuildingSegmentRule(),
+                index_rmd=BASELINE_0,
+                list_path="buildings[*].building_segments[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+
+            return {"rmd_b": rmd_b}
+
+        class BuildingSegmentRule(RuleDefinitionBase):
```

Comment:
My take is this rule should be at `service_water_heating_distribution_systems` level, not building level. (See RDS Evaluation Context)

In RMD Rule, we need to add an `is_applicable` to check if there are `service_water_heating_distribution_systems` or not.
In RMD Rule `create_data`, we should call `get_swh_components_associated_with_each_swh_distribution_system` to get the data.

In the ServiceWaterHeatingDistributionSystem, we need to check every piping, including children, is modeled with thermal losses turned to false.
You can reuse the same code block searches for piping under a ServiceWaterHeatingDistributionSystem from the `get_swh_components_associated_with_each_swh_distribution_system` function.